### ISSUE TITLE
Tensorflow Debug tfdbg ValueError with combined loss functions.

### ISSUE THREAD

**Issue by zazabar** [tensorflow/tensorflow#16261]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes. Included
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 Home x64, 1709
- **TensorFlow installed from (source or binary)**: binary (pip3)
- **TensorFlow version (use command below)**:  1.4.0
- **Python version**:  3.5.2
- **Exact command to reproduce**: See source code.

### Describe the problem
When attempting to combine two loss functions, tfdbg fails to properly grab the gradients due to a ValueError when executing run.

### Source code / logs
[consolelog.txt](https://github.com/tensorflow/tensorflow/files/1649424/consolelog.txt) Log of the error.
[not_working.txt](https://github.com/tensorflow/tensorflow/files/1649426/not_working.txt) Code example that generates the error.
[working.txt](https://github.com/tensorflow/tensorflow/files/1649427/working.txt) Code example where the two loss functions are split that does not generate the error.

**Comment by caisq** [comment]
@zazabar Thanks for taking the time to report this issue and creating the example code for reproduction.

However, I can't reproduce the issue using your `not_working.txt` on a GPU machine that I have. See screenshot of the successful `tfdbg> run` below:
![tfdbg_bug_fail_bug_works](https://user-images.githubusercontent.com/16824702/35258674-141b22aa-ffcf-11e7-8f65-62bc792f6b27.png)

The `tfdbg> run` passed 5 out of the 5 times that I ran it. This was the case for the latest nightly version of tensorflow-gpu (1.5). I didn't try 1.4 though (too painful to downgrade CUDA from 9 to 8... :p )

I wonder whether this has to do with GPU type or OS details.
Mine is a GeForce GTX 1050 Ti. My OS is Ubuntu 16.04.

I haven't tried Windows, though. Maybe this is a Windows-specific issue. If you could try a Linux machine, that would be great. I can also try a Windows machine with GPU later.

**Comment by zazabar** [comment]
Hey @caisq ,

As you indicated, when I ran it on a Linux machine (Mint), it did not run into any problems. So perhaps this is just a Windows issue. Is there anything else that I can supply that might help fish out what the culprit is?

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
